core/cont/src/TArrayC.cxx

```diff
@@ -109,13 +109,15 @@ void TArrayC::Set(Int_t n)
       Char_t *temp = fArray;
       if (n != 0) {
          fArray = new Char_t[n];
-         if (n < fN) memcpy(fArray,temp, n*sizeof(Char_t));
-         else {
-            memcpy(fArray,temp,fN*sizeof(Char_t));
+         if (n < fN) {
+            memcpy(fArray, temp, n*sizeof(Char_t));
+         } else {
+            if (temp)
+               memcpy(fArray, temp, fN*sizeof(Char_t));
             memset(&fArray[fN],0,(n-fN)*sizeof(Char_t));
```

Comment:
I suspect this might be a better handling of the "exceptional" condtion.
```suggestion
         } else if (temp) {
            memcpy(fArray, temp, fN*sizeof(Char_t));
            memset(&fArray[fN],0,(n-fN)*sizeof(Char_t));
         } else {
            memset(&fArray[0],0,fN*sizeof(Char_t));         
         }
```

Reply:
Actually, one should do:
```
memset(fArray,0,n*sizeof(Char_t)); 
```
I submit correspondent changes